src/NuGetPackageVerifier/PackageAnalysisContext.cs

```diff
@@ -18,9 +21,18 @@ public class PackageAnalysisContext : IDisposable
         public IPackageVerifierLogger Logger { get; set; }
         public PackageArchiveReader PackageReader => _reader ?? (_reader = new PackageArchiveReader(PackageFileInfo.FullName));
 
+        public IDictionary<string, AssemblyAttributesData> AssemblyData = new Dictionary<string, AssemblyAttributesData>();
+
         public void Dispose()
         {
             _reader?.Dispose();
         }
     }
+
+    public class AssemblyAttributesData
+    {
+        public AssemblyDefinition Assembly { get; set; }
```

Comment:
To get `AssemblyVersion` [here](https://github.com/aspnet/BuildTools/blob/jbagga/StatelessNPV166/src/NuGetPackageVerifier/Rules/PackageVersionMatchesAssemblyVersionRule.cs#L49) since it isn't a custom attribute